airflow/serialization/enums.py

```diff
@@ -50,3 +50,4 @@ class DagAttributeTypes(str, Enum):
     PARAM = 'param'
     XCOM_REF = 'xcomref'
     DATASET = 'dataset'
+    ENCODING_ENUM = '__encoding_enum__'
```

Comment:
Why not just `ENCODING`? (since the class is Encoding)

Reply:
Well, we're probably not going to implement this anyway so maybe moot.... but here's why....

the only time this is used is for serializing an already-serialized object.

that means serializing a dict whos keys are the special Encoding enum.  the tricky thing is we can't just encode these keys in the normal way, because that would mean replacing the string key with a dictionary, and a dictionary is not hashable and can't be used as a key in a dict.  we can't make it a tuple either because a tuple as a dict key is not json-serializable.  so we have to make it a string prefix.  so, we could have done `encoding-__var` but... when we deserialize, we're looking for that prefix... with `encoding-` as the prefix, it seems we'd get a lot more false positives than with `__encoding_enum__`  .... i guess `__encoding__` would be fine if that's what you're suggesting....  i dunno wdyt?

Reply:
Should we make it a bit simpler ? do we actually need to serialize an already serialized object? I guess it would be simpler if we add a __serialized__= True attribute in the whole object and skip serialization in this case altogether? Unless we want to handle a recursive serialize/deserialize in which case it should be __serialization_depth__: Int possibly :)

But I think it should rathere be handled via proper "flow" of serialization and avoiding double or multiple serialization (which I guess what you mean by " not going to implement it anyway" ?